airflow/hooks/base_hook.py

```diff
@@ -90,3 +93,78 @@ def get_hook(cls, conn_id: str) -> "BaseHook":
     def get_conn(self) -> Any:
         """Returns connection for the hook."""
         raise NotImplementedError()
+
+
+class DiscoverableHook(Protocol):
+    """
+    This protocol Just describes what Hook classes that want to be discovered by ProvidersManager can
+    implement. It is not used by any of the Hooks, but simply methods and class fields described here
+    are implemented by those Hooks.
```

Comment:
```suggestion
    Interface that providers *can* implement to be discovered by ProvidersManager.
    
    It is not used by any of the Hooks, but simply methods and class fields described here are
    implemented by those Hooks. Each method is optional -- only implement the ones you need.
```

Reply:
Main reason is to keep the "short description" (up to the first full stop) short.

Reply:
Goodl